### ISSUE TITLE
GameImage GUI not finding exe and/or looking at the wrong location

### ISSUE THREAD

**Issue by tanner02** [ruanformigoni/gameimage#6]
Title is pretty much self-explanatory. Even if i put the executable into that folder (for test purposes obviously) it still doesn't appear as a selectable file.
![Screenshot_20230122_132134](https://user-images.githubusercontent.com/48131064/213915609-bb9e2933-7045-4d9e-8fb2-5cfb15152740.png)

**Comment by ruanformigoni** [comment]
Hi @tanner02, thanks for the bug report! I fixed the issue in version `v0.3.4`, it should now only show the `.exe` or `.msi` files.

**Comment by tanner02** [comment]
Hi @ruanformigoni! Thank you for the quick fix once again. Unfortunately the "rom" button is [non-functional now](https://tanner.discowd.com/r/2023-01-22-17_09_41.548450592.gif).

**Comment by ruanformigoni** [comment]
From the gif, are there executable files in the directory/subdirectories? Also, could you try to run like that anyway, to see if it works?

**Comment by tanner02** [comment]
The executable is present in `ProStreet/rom/nfs.exe` as well as others such as the uninstaller. Trying to build it anyways results in [this](https://tanner.tixte.co/r/Screenshot_20230122_182630.png).

**Comment by ruanformigoni** [comment]
Ah, you do not have `yq` installed in your system, I think this causes both problems. I've embed it into the appimage in a [new version](https://mega.nz/file/N1wBFCQQ#ycns1b1Bub7oX8F5Yv_D3Oz-67xWCF94JZ4Ps6MFlg8), could you please test with it again?

**Comment by ruanformigoni** [comment]
Hi @tanner02, I've updated the gui with `yq` and a fix for broken output in the gui in version `v0.3.7`. See if that also fixes the issue with the `invalid platform` message. Also please try to erase `/home/deck/Downloads/gameimage.yml`.

**Comment by tanner02** [comment]
The build uploaded on mega.nz and version 0.3.7 from gitlab successfully build the appimage. However the "rom" button is still non-functional on my end, which i guess isn't a big deal when it builds the appimage just fine. Despite the successful build i've noticed 2 errors while the program was processing stuff.

> /tmp/.mount_GameImoya3T6/usr/bin/common.sh: line 31: [: : integer expression expected

**Comment by ruanformigoni** [comment]
Thanks you so much for your feedback, this:
> /tmp/.mount_GameImoya3T6/usr/bin/common.sh: line 31: [: : integer expression expected
Was the same issue as before, with other check in the codebase I forgot to correct.  Gitlab is processing `0.3.8`, this should fix it. I'll try to replicate the issue with the rom button.

**Comment by ruanformigoni** [comment]
Hi @tanner02 , I was able to replicate the issue with a `symlink`, the gui was not following them. I've added release `0.3.9`, gitlab is currently processing.

**Comment by tanner02** [comment]
Hi @ruanformigoni, thanks a lot for your hard work! I hate to break it to you but the rom button still doesn't work on my end 😅
I've build an appimage with 0.3.9 now and i'm glad to report that i haven't encountered any errors. Except,

> [*] AppRun written, make further changes to it if you desire, then press enter...
/tmp/.mount_imagemtiUKUO/AppRun: line 22: [: too many arguments

**Comment by ruanformigoni** [comment]
Haha, no problem! Does it have any output on the terminal, if you run with `$ ./GameImage-x86_64.AppImage` (for the gui)?

For this problem, I've create a [debug build](https://mega.nz/file/thJWwRqT#-UxJUTguXedbe6EAhYGOM5lkX29UDeARUvqdKM4otmU), this should show each command that is executed, before it is executed, to figure out what caused the message.
> [*] AppRun written, make further changes to it if you desire, then press enter...
> /tmp/.mount_imagemtiUKUO/AppRun: line 22: [: too many arguments

**Comment by ruanformigoni** [comment]
I've managed to find the issue, it was really obvious, just me being dumb :smiling_face_with_tear: . Gitlab is processing `v0.4.0` with the hotfix. If it still doesn't show up, let me know. Thanks for helping me improve the software!

**Comment by tanner02** [comment]
Ah man thank you! No need to be harsh on yourself it was just a human error. I'm very happy to report that it **works perfectly** now! The button as well as the .exe detection. I'm glad to somewhat help on improving the software!

About the output: There is no output on the terminal when running the GUI there.

I've run the debug build, the `too many arguments` error appears to be program/game specific. It doesn't occur when packaging NFS ProStreet but when packaging another game like "Ridge Racer Unbounded". I don't know if the error affects the functionality of the appimage because i haven't figured out yet how to properly install DirectX/Vcredist inside the package (wine fails to install them) which i think is the reason those 2 games i've packaged don't launch. Either way, here's a snippet of the command lines before and after the error code.

> + path_exec='wine/drive_c/Program Files (x86)/Ridge Racer Unbounded/RRU.exe'
> ++ basename 'wine/drive_c/Program Files (x86)/Ridge Racer Unbounded/RRU.exe'
> + sed -E 's/^\s+://'
> + tee AppDir/AppRun
> #!/usr/bin/env bash
> 
> set -e
> 
> # Wine executable
> WINE="$APPDIR/usr/bin/wine"
> 
> if [[ "$(basename "${APPIMAGE}")" =~ \.\.AppImage ]]; then
>   # Set hidden config dir
>   CFGDIR="$(dirname "$APPIMAGE")/$(basename "$APPIMAGE").config"
> else
>   # Set visible config dir
>   CFGDIR="$(dirname "$APPIMAGE")/.$(basename "$APPIMAGE").config"
> fi
> 
> # Path to appimage mountpoint
> MNTDIR="$APPDIR"
> 
> # Create wine prefix if not exists
> export WINEPREFIX="$CFGDIR/wine"
> if [ ! -d "$WINEPREFIX" ]; then
>   mkdir -p "$CFGDIR"
>   cp -r "$MNTDIR/app/wine" "$CFGDIR"
> fi
> 
> # Create/Update symlink to the application directory
> rm -f "$CFGDIR/wine/drive_c/Program Files (x86)"
> ln -s "$MNTDIR/app/rom/Program Files (x86)" "$CFGDIR/wine/drive_c/Program Files (x86)"
> 
> cd "$(dirname "$CFGDIR/wine/drive_c/Program Files (x86)/Ridge Racer Unbounded/RRU.exe")"
> 
> exec="RRU.exe"
> 
> if [ "$@" ]; then
>   "$WINE" "$@"
> else
>   "$WINE" "$exec"
> fi
> + chmod +x AppDir/AppRun
> + msg -n 'AppRun written, make further changes to it if you desire, then press enter...'
> + declare -a opts
> + '[' 2 -gt 1 ']'
> + opts+=("$1")
> + shift
> + '[' 1 -gt 1 ']'
> + '[' -z Yes ']'
> + echo -n '[*] AppRun written, make further changes to it if you desire, then press enter...'
> [*] AppRun written, make further changes to it if you desire, then press enter...+ read -r
> + ./imagemagick '/run/media/mmcblk0p1/Games/Ridge Racer Unbounded/icon/25d8a1b90aaf912eaa8831c2baf0f055.png' AppDir/ridge.png
> /tmp/.mount_imagem4Om00B/AppRun: line 22: [: too many arguments
> + desktop_entry_create ridge
> + local name
> + name=ridge
> + name=ridge
> + declare -a name_alt
> + for i in $name
> + name_alt+=("${i^}")
> + name=ridge
> + sed -E 's/^\s+://'
> + tee AppDir/ridge.desktop
> [Desktop Entry]
> Name=Ridge
> Exec=/usr/bin/bash
> Icon=ridge
> Type=Application
> Categories=Utility;
> + appdir_build
> + ARCH=x86_64
> + ./appimagetool AppDir
> appimagetool, continuous build (commit b719a7f), build <local dev build> built on 2022-09-29 13:55:16 UTC

**Comment by ruanformigoni** [comment]
I'm really happy that it is working :smile: . Thanks to your log, I figured out the issue, it is `imagemagick's` fault, it doesn't handle spaces in paths properly. However, it should still work as intended, generating the icon in appdir, here is a [version that I patched](https://mega.nz/file/ZgwzgQaQ#m8DgEWki4PhLG0Qy8S3I9_daWrlipvRmLmOxQ6CH2Vk) and it should fix the warning. I'll create a fork of it with my fix to use it instead, and create an issue in the imagemagick project. About your games not running, could you post the terminal output? Maybe I could help with that.

**Comment by tanner02** [comment]
Glad you figured it out. I will give the patched version a try, thank you.

About the games:
I get a `terminate called after throwing an instance of 'dxvk::DxvkError'`

Possibly unrelated but here are the logs of when trying to install directx and vcredist 2008:

DirectX:

> wine: using kernel write watches (experimental).
> wineserver: using server-side synchronization.
> wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> 0120:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
> 0120:err:ntlm:ntlm_LsaApInitializePackage no NTLM support, expect problems
> 0118:err:mscoree:LoadLibraryShim error reading registry key for installroot
> 0118:err:mscoree:LoadLibraryShim error reading registry key for installroot
> 0130:err:mscoree:LoadLibraryShim error reading registry key for installroot

Vcredist:

> wine: using kernel write watches (experimental).
> wineserver: using server-side synchronization.
> wine: RLIMIT_NICE is <= 20, unable to use setpriority safely
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> 0148:err:setupapi:SetupDefaultQueueCallbackW copy error 1812 L"@C:\\windows\\system32\\drivers\\wineusb.sys,-1" -> L"C:\\windows\\inf\\wineusb.inf"
> wine: using kernel write watches (experimental).
> 0150:err:appwizcpl:install_file MsiInstallProduct failed: 1627
> 0148:err:advpack:create_tmp_ini_file Can't get temp ini file path
> 0148:err:mshtml:register_server RegInstall failed: 80004005
> 0148:err:advpack:create_tmp_ini_file Can't get temp ini file path
> wine: using kernel write watches (experimental).
> 0158:err:advpack:create_tmp_ini_file Can't get temp ini file path
> wine: using kernel write watches (experimental).
> 0168:err:advpack:create_tmp_ini_file Can't get temp ini file path
> 0168:err:mshtml:register_server RegInstall failed: 80004005
> 0168:err:advpack:create_tmp_ini_file Can't get temp ini file path
> wine: using kernel write watches (experimental).
> 0170:err:advpack:create_tmp_ini_file Can't get temp ini file path
> 0024:err:ver:GetFileVersionInfoSizeExW HACK: not exposing version info.
> 0024:err:ver:GetFileVersionInfoSizeExW HACK: not exposing version info.
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> wine: using kernel write watches (experimental).
> 01a4:err:kerberos:kerberos_LsaApInitializePackage no Kerberos support, expect problems
> 01a4:err:ntlm:ntlm_LsaApInitializePackage no NTLM support, expect problems

Thank you for your help!

**Comment by ruanformigoni** [comment]
I've packaged ProStreet, usually, you do not need to install directx and vcredist. This is an issue with `dxvk`, are you on a steam deck?

**Comment by tanner02** [comment]
Yep, i am.

**Comment by ruanformigoni** [comment]
It is probably an issue with missing 32-bit drivers, have you tried to install these games in a 64-bit prefix?

**Comment by tanner02** [comment]
Thanks a lot for the help! I've used the wine image you've linked and now ProStreet booted perfectly fine! :smile:

**Comment by ruanformigoni** [comment]
Nice, did you need to install the drivers?

**Comment by tanner02** [comment]
Nope, i would have installed them if it didn't work though.

**Comment by ruanformigoni** [comment]
Thanks a lot for your feedback! I'll close this issue for now, if you have any more issues let me know.

**Comment by ruanformigoni** [comment]
Forgot to mention, I've set the appimage that worked as the default in the latest release. The apparently broken ones were embedding `32-glibc`, which doesn't seem to be as good as using `wow64`. These wine issues should be resolved in future releases.

**Comment by tanner02** [comment]
Thanks for the heads up! I was about to manually replace it by myself like i did with ProStreet :sweat_smile:

**Comment by tanner02** [comment]
Hi @ruanformigoni, its me again. I've got a short question. How can i run winecfg while building the appimage to e.g. add dll overrides to it?

**Comment by ruanformigoni** [comment]
Hey! It was a bit of a hassle to do it, so I made a change in [this build](https://mega.nz/file/RxpGkZqI#uoVXJ_3kNDHn299zXHIOLesBx_5iUslwKpa_AwzZFWs), tell me if it works, so I'll turn it into a new release. Thanks

**Comment by tanner02** [comment]
You're awesome dude! Sorry for the delayed reply. I've tried it and can confirm that it works. There's one issue though: [Wine fails to find the rom folder if there's space in the path](https://tanner.tixte.co/r/Screenshot_20230126_155413.png). Changing `/run/media/mmcblk0p1/Games/pro street/` to `/run/media/mmcblk0p1/Games/prostreet/` worked as a temporary fix when testing. But besides that, everything works as it should. Thanks a lot again and i'm sorry if these requests put you under stress.

**Comment by ruanformigoni** [comment]
Thanks! No problem :smile: . Feel free to make requests, during weekdays I might take a bit longer to answer; nice catch with the spaces problem, I fixed it and released it `0.4.2`.

**Comment by tanner02** [comment]
Aw thank you man! It's the most i can do :sweat_smile:. Thanks for the fix, I'll let you know if i find another issue or request!

**Comment by ruanformigoni** [comment]
Imagemagick merged my [pull request](https://github.com/ImageMagick/ImageMagick/pull/5993), should resolve the previously found issue in paths that contain spaces.
